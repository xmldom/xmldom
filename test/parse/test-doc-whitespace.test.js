'use strict';

const { describe, expect, test } = require('@jest/globals');
const { MIME_TYPE } = require('../../lib/conventions');
const { getTestParser } = require('../get-test-parser');

describe('errorHandle', () => {
	test('document source', () => {
		const testSource = '<?xml version="1.0"?>\n<!--test-->\n<xml/>';
		const { errors, parser } = getTestParser();

		const actual = parser.parseFromString(testSource, MIME_TYPE.XML_TEXT).toString();

		expect({ actual, ...(errors.length ? { errors } : undefined) }).toMatchSnapshot();
	});

	test('should encode < literal when not part of a tag', () => {
		const description = '<p>populaciji (< 0.1%), te se</p>';
		const { errors, parser } = getTestParser();

		const actual = parser.parseFromString(description, 'text/html').toString();

		expect({ actual, ...(errors.length ? { errors } : undefined) }).toMatchSnapshot();
	});
});
