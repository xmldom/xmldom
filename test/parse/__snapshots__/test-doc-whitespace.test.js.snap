// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`errorHandle document source 1`] = `
Object {
  "actual": "<?xml version=\\"1.0\\"?>
<!--test-->
<xml/>",
}
`;

exports[`errorHandle should encode < literal when not part of a tag 1`] = `
Object {
  "actual": "<p xmlns=\\"http://www.w3.org/1999/xhtml\\">populaciji (&lt; 0.1%), te se</p>",
  "errors": Array [
    Array [
      "error",
      "element parse error: Error: invalid tagName:",
      Object {
        "columnNumber": 17,
        "lineNumber": 1,
      },
    ],
  ],
}
`;
